localstack-core/localstack/services/sns/v2/provider.py

```diff
@@ -118,6 +164,360 @@ def set_topic_attributes(
             raise InvalidParameterException("Invalid parameter: AttributeName")
         topic["attributes"][attribute_name] = attribute_value
 
+    ## Subscribe operations
+
+    def subscribe(
+        self,
+        context: RequestContext,
+        topic_arn: topicARN,
+        protocol: protocol,
+        endpoint: endpoint | None = None,
+        attributes: SubscriptionAttributesMap | None = None,
+        return_subscription_arn: bool | None = None,
+        **kwargs,
+    ) -> SubscribeResponse:
+        parsed_topic_arn = parse_and_validate_topic_arn(topic_arn)
+        if context.region != parsed_topic_arn["region"]:
+            raise InvalidParameterException("Invalid parameter: TopicArn")
+
+        store = self.get_store(account_id=parsed_topic_arn["account"], region=context.region)
+
+        if topic_arn not in store.topic_subscriptions:
```

Comment:
nit: this should now be `if topic_arn not in store.topics` 

We should try to unify usage of `topics` around the codebase and avoid having multiple attributes for the same resource


could be (for future usage in this same function)

```python
if topic_arn not in store.topics:
    raise NotFoundException("Topic does not exist")

topic_subscriptions = store.topics[topic_arn]["subscriptions"]
```

and then `topic_subscriptions` can be used through this function